addons/hr/models/hr_employee.py

```diff
@@ -125,11 +123,34 @@ class HrEmployeePrivate(models.Model):
         ('user_uniq', 'unique (user_id, company_id)', "A user cannot be linked to multiple employees in the same company.")
     ]
 
-    def _get_placeholder_filename(self, field=None):
-        image_fields = ['image_%s' % size for size in [1920, 1024, 512, 256, 128]]
-        if field in image_fields:
-            return 'hr/static/src/img/default_image.png'
-        return super()._get_placeholder_filename(field=field)
+    @api.depends('name', 'user_id.share', 'image_1920')
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends('name', 'user_id.share', 'image_1024')
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends('name', 'user_id.share', 'image_512')
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends('name', 'user_id.share', 'image_256')
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends('name', 'user_id.share', 'image_128')
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _get_one_avatar(self, avatar_field, image_field):
+        self.ensure_one()
+        if self.user_id and not self.user_id.share:
+            return super()._generate_avatar_svg(self[self._rec_name])
+        return self._get_avatar_placeholder()
+
+    def _get_avatar_seed(self):
+        return str(self.user_id.create_date.timestamp()) or super()._get_avatar_seed()
```

Comment:
This `or` is confusing, I would be expecting block conditions "if user: return user date; return super"
Doesn't this even crash to call timestamp on create_date is user_id is False?